frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -438,6 +470,169 @@ pub mod pallet {
+        /// entire staked amount will be unstaked.
+        ///
+        /// If user attempts to unstake more than is available, everything will be unstaked.
+        ///
+        /// The dispatch origin for this call must be _Signed_ by the staker's account.
+        ///
+        /// Note that all funds will remain staked after this call, only the unbonding process is started.
+        /// Staker will still earn rewards as long as funds aren't withdrawn.
+        ///
+        #[pallet::weight(1000_0000)]
+        pub fn start_unbonding(
+            origin: OriginFor<T>,
+            contract_id: T::SmartContract,
+            #[pallet::compact] value: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            let staker = ensure_signed(origin)?;
+
+            ensure!(value > Zero::zero(), Error::<T>::UnstakingWithNoValue);
+            ensure!(
+                Self::is_active(&contract_id),
+                Error::<T>::NotOperatedContract,
+            );
+
+            // Get the unlocking chunks for this staker and contract
+            let mut unbonding_info = UnbondingInfoStorage::<T>::get(&staker, &contract_id);
+            ensure!(
+                unbonding_info.len() < T::MaxUnlockingChunks::get(),
+                Error::<T>::TooManyUnlockingChunks
+            );
+            let unlocking_chunks_sum = unbonding_info.sum();
+
+            // Get the latest era staking points for the contract.
+            let current_era = Self::current_era();
+            let staking_info = Self::staking_info(&contract_id, current_era);
+            ensure!(
+                staking_info.stakers.contains_key(&staker),
+                Error::<T>::NotStakedContract,
+            );
+            let staked_value = staking_info.stakers[&staker];
+
+            // Calculate the value which can be unstaked.
+            let value_available_for_unstaking = staked_value.saturating_sub(unlocking_chunks_sum);
+            let remaining_value_after_unstaking =
+                value_available_for_unstaking.saturating_sub(value);
+
+            // In case we would go below the minimum staking amount, unstake everything
+            let value_to_unstake =
+                if remaining_value_after_unstaking < T::MinimumStakingAmount::get() {
+                    value_available_for_unstaking
+                } else {
+                    value.min(value_available_for_unstaking)
+                };
+
+            // It is possible that the entire value is already in the process of being unstaked.
+            ensure!(
+                value_to_unstake > Zero::zero(),
+                Error::<T>::UnstakingWithNoValue
+            );
+
+            // TODO: NOTE that UnbondingPeriod doesn't mean the full eras. E.g. if it is set to 1, it will expire after current era.
+            // Should we add 1 to ensure we get number of full eras?
+
+            // Update the chunks and write them to storage
+            unbonding_info.push(UnlockingChunk {
```

Comment:
I want to confirm the functionality;
- So users are able to submit multiple unbonding requests with different time intervals?
- Rewards will not be counted for unbonding stakes?
- After the unbonding period, the user needs to submit with `unstake_and_withdraw`?

Reply:
Users can submit up to **MaxUnlockingChunks** requests per contract.
They cannot submit more until they've withdrawn some of the previous ones.

Actually, as it is now, the part that's undergoing unbonding process will be counted until user withdraws it.

Correct, after the unbonding period, user needs to unlock/withdraw the chunks that have gone through the unbonding period.

Regarding the first point, I assume there was a misunderstanding.
I will also reply in Slack.